src/internaltypes.h

```diff
@@ -22,6 +22,13 @@ limitations under the License.
 #include "libcellml/types.h"
 #include "libcellml/variable.h"
 
+#undef NAN
+
+#ifdef __linux__
+#    undef TRUE
+#    undef FALSE
+#endif
```

Comment:
Why are you wrapping `TRUE` and `FALSE` but not `NAN`?

Reply:
Simply being `NAN` is originally defined on all our supported platforms while `TRUE` and `FALSE` are only originally defined on Linux...? I will confirm this later today.

Reply:
I have just checked on Windows, Linux and macOS and, indeed, `NAN` needs to be undefined on all three platforms while `TRUE` and `FALSE` only need to be undefined on Linux.